### PR TITLE
chore(): refactor

### PR DESCRIPTION
**Pull request by tvcam**
Here is my first refactor, as well as reviewing. Don't hesitate to question me, I'll also comment some code to explain why I change it.

I follow
- https://github.com/bbatsov/ruby-style-guide
- https://www.sitepoint.com/10-ruby-on-rails-best-practices/
- SOLID principle
- DRY principle

### CODE REVIEW

File: app/controllers/movies_controller.rb [hunk 1]
```diff
@@ -4,27 +4,21 @@ class MoviesController < ApplicationController
...
-  end
-
-  def unrated
-    @movies = Movie.unrated
-  end
 
-  def latest
-    @movies = Movie.latest
+  # sort actions
+  def sort
```
Comment (tvcam) [inline app/controllers/movies_controller.rb:16]: `rating`, `unrated` and `latest` are kind of `sorting`, so I create a sort action instead (DRY: Don't Repeat Yourself ).

File: app/controllers/movies_controller.rb [hunk 2]
```diff
@@ -79,17 +70,13 @@ def update
   # DELETE /movies/1
   # DELETE /movies/1.json
   def destroy
-    @movie.delete_all_reviews
```
Comment (tvcam) [inline app/controllers/movies_controller.rb:82]: I added `has_many :reviews, dependent: :destroy` in `Movie`, so when we call `@movie.destroy`, the associate reviews also destroyed [More](http://apidock.com/rails/v4.2.1/ActiveRecord/Associations/ClassMethods/has_many)

File: app/controllers/quotes_controller.rb [hunk 3]
```diff
@@ -1,26 +1,18 @@
 class QuotesController < ApplicationController
-
-
   def create
-
-    @quote = Quote.create(quote_params)
```
Comment (tvcam) [inline app/controllers/quotes_controller.rb:6]: `create` is `new` then `save` so the block here is repeated by calling `save` twice

File: app/controllers/reviews_controller.rb [hunk 4]
```diff
@@ -1,14 +1,12 @@
 class ReviewsController < ApplicationController
-  before_action :set_movie_and_review, only: [:show, :edit, :update, :destroy]
+  before_action :set_movie, except: [:user_reviews]
```
Comment (tvcam) [inline app/controllers/reviews_controller.rb:2]: SRP: Single Responsibility Principle, don't do much inside one method.

File: app/models/movie.rb [hunk 5]
```diff
@@ -1,27 +1,34 @@
 class Movie < ActiveRecord::Base
-  belongs_to :user
-  has_many :reviews
-  has_many :users, through: :reviews
-  has_many :quotes
-  validates :title, presence: true
-  validates :synopsis, presence: true
+  belongs_to  :user
```
Comment (tvcam) [inline app/models/movie.rb:2]: Alignment & Indentation is for readability, you've better get used to it lol

File: app/models/movie.rb [hunk 6]
```diff
@@ -1,27 +1,34 @@
...
     format: {
       with: /(19|20)\d{2}/i,
-      message: "should be a four-digit year"
+      message: 'should be a four-digit year'
     }
-    accepts_nested_attributes_for :reviews, reject_if: :all_blank
 
+  accepts_nested_attributes_for :reviews, reject_if: :all_blank
+
+  scope :sorted_by_rating,  -> { order(rating: :desc) }
```
Comment (tvcam) [inline app/models/movie.rb:18]: Prefer scope to class method for simple thing.

File: app/models/movie.rb [hunk 7]
```diff
@@ -1,27 +1,34 @@
...
+  scope :sorted_by_year,    -> { order(year: :desc) }
 
+  scope :rateds,  -> { where.not(rating: nil) }
+  scope :unrated, -> { where(rating: nil) }
 
   def reviews_attributes=(attributes)
```
Comment (tvcam) [inline app/models/movie.rb:24]: Will refactor this method later, this is quite messy, no?

File: app/models/review.rb [hunk 8]
```diff
@@ -1,9 +1,9 @@
 class Review < ActiveRecord::Base
-  belongs_to :user
-  belongs_to :movie
-  validates :title, presence: true
-  validates :content, presence: true
+  belongs_to  :user
+  belongs_to  :movie
 
-  validates_numericality_of :rating, :greater_than_or_equal_to => 0.0, :less_than_or_equal_to => 10.0, precision: 1
+  validates   :title,
```
Comment (tvcam) [inline app/models/review.rb]: Prefer new syntax `validates` with options

File: app/models/user.rb [hunk 9]
```diff
@@ -9,12 +9,11 @@ class User < ActiveRecord::Base
   has_many :movies
 
   def self.from_omniauth(auth)
-    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
-        user.provider = auth.provider
-        user.uid = auth.uid
-        user.email = auth.info.email
-        user.password = Devise.friendly_token[0,20]
-      end
+    find_or_create_by(provider: auth.provider, uid: auth.uid) do |user|
```
Comment (tvcam) [inline app/models/user.rb:12]: As method name suggested, it first find the record, if not found then create.

File: app/views/movies/_index.html.erb [hunk 10]
```diff
@@ -1,49 +1,57 @@
 <p id="notice"><%= notice %></p>
 
 <h1>List of Movies in the DB</h1>
-<%= link_to "Sort by rating (unrated not included)", movies_rating_url %> | <%= link_to "Show all unrated films", movies_unrated_url %>
-| <%= link_to "Sort by Year (latest first)", movies_latest_url %>
+<%= link_to "Sort by rating (unrated not included)", sort_movies_path(rating: true) %> | <%= link_to "Show all unrated films", sort_movies_path(unrated: true) %>
```
Comment (tvcam) [inline app/views/movies/_index.html.erb:4]: change according to `sort` action.

File: app/views/movies/_index.html.erb [hunk 11]
```diff
@@ -1,49 +1,57 @@
...
-    <tr>
-      <th>Title</th>
-      <th>Year</th>
-      <th>Synopsis</th>
-      <th>Rating</th>
-      <th>No. of Reviews</th>
-      <th colspan="3"></th>
-    </tr>
-  </thead>
+  <%# Create a table for movie iteration %>
```
Comment (tvcam) [inline app/views/movies/_index.html.erb:7]: Please code with proper alignment, it help a lot to know what the code doing, editor can help just tap tap.

File: app/models/review.rb [hunk 12]
```diff
@@ -1,9 +1,9 @@
...
+              :content, presence: true
 
+  validates   :rating, numericality: { greater_than_or_equal_to: 0.0, less_than_or_equal_to: 10.0, precision: 1 }
```
Comment (tvcam) [inline app/models/review.rb:8]: Prefer new syntax `validates` with options

File: app/models/movie.rb (hunk 6 above)
Comment (jakeekaj) [inline app/models/movie.rb:18]: Kindly elaborate on this please. :)
Thanks!

Comment (tvcam) [inline app/models/movie.rb:18]: Scope is chainable and easy to handle returning objects. A nice explanation here: http://www.justinweiss.com/articles/should-you-use-scopes-or-class-methods/, please look at the big header `Why use scopes when we already have class methods?`

Comment (jakeekaj) [inline app/models/movie.rb:18]: Thanks! Got it.

**Pull request merged**